src/praisonai/praisonai/bots/_identity.py

```
@@ -0,0 +1,182 @@
+            explicit = self._links.get((platform, platform_user_id))
+        if explicit is not None:
+            return explicit
+
+        # 2. Pairing-store read-through: a paired channel's label is the
+        #    canonical id shared across channels paired under it.
+        if self._pairing_store is not None and self._use_pairing_label:
+            try:
+                canonical = self._canonical_from_pairing(platform, platform_user_id)
+                if canonical:
+                    return canonical
+            except Exception as e:  # pragma: no cover — defensive
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)